### PR TITLE
Initial check in for document db.

### PR DESCRIPTION
**Pull request by alvadb**

### CODE REVIEW

File: azure-mgmt-documentdb/src/main/java/com/microsoft/azure/management/documentdb/implementation/DatabaseAccountImpl.java [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The implementation for DatabaseAccount.
+ */
+@LangDefinition
+public class DatabaseAccountImpl
```
Comment (jianghaolu) [inline azure-mgmt-documentdb/src/main/java/com/microsoft/azure/management/documentdb/implementation/DatabaseAccountImpl.java]: internal

File: azure-mgmt-documentdb/src/main/java/com/microsoft/azure/management/documentdb/implementation/DatabaseAccountImpl.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+                DatabaseAccountInner,
+                DatabaseAccountImpl,
+                DocumentDBManager>
+        implements DatabaseAccount,
+        DatabaseAccount.Definition,
+        DatabaseAccount.Update {
+    private List<Location> allLocations;
+    private Map<String, FailoverPolicyInner> failoverPolicies;
+
+    protected DatabaseAccountImpl(String name, DatabaseAccountInner innerObject, DocumentDBManager manager) {
```
Comment (jianghaolu) [inline azure-mgmt-documentdb/src/main/java/com/microsoft/azure/management/documentdb/implementation/DatabaseAccountImpl.java]: internal

File: azure-mgmt-documentdb/src/main/java/com/microsoft/azure/management/documentdb/implementation/DatabaseAccountsImpl.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+import rx.Completable;
+import rx.Observable;
+import rx.functions.Func1;
+
+import java.util.List;
+
+/**
+ * Implementation for Registries.
+ */
+public class DatabaseAccountsImpl
```
Comment (jianghaolu) [inline azure-mgmt-documentdb/src/main/java/com/microsoft/azure/management/documentdb/implementation/DatabaseAccountsImpl.java]: internal

File: azure-mgmt-documentdb/src/main/java/com/microsoft/azure/management/documentdb/implementation/DatabaseAccountsImpl.java [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        extends
+        GroupableResourcesImpl<
+                DatabaseAccount,
+                DatabaseAccountImpl,
+                DatabaseAccountInner,
+                DatabaseAccountsInner,
+                DocumentDBManager>
+        implements DatabaseAccounts {
+
+    protected DatabaseAccountsImpl(final DocumentDBManager manager) {
```
Comment (jianghaolu) [inline azure-mgmt-documentdb/src/main/java/com/microsoft/azure/management/documentdb/implementation/DatabaseAccountsImpl.java]: internal

File: azure-mgmt-documentdb/src/main/java/com/microsoft/azure/management/documentdb/implementation/DocumentDBManager.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+    /**
+     * Creates an instance of ComputeManager that exposes Compute resource management API entry points.
+     *
+     * @param credentials the credentials to use
+     * @param subscriptionId the subscription
+     * @return the ComputeManager
+     */
+    public static DocumentDBManager authenticate(AzureTokenCredentials credentials, String subscriptionId) {
+        return new DocumentDBManager(new RestClient.Builder()
```
Comment (jianghaolu) [inline azure-mgmt-documentdb/src/main/java/com/microsoft/azure/management/documentdb/implementation/DocumentDBManager.java]:
```java
        return new DocumentDBManager(new RestClient.Builder()
                .withBaseUrl(credentials.environment(), AzureEnvironment.Endpoint.RESOURCE_MANAGER)
                .withCredentials(credentials)
                .withSerializerAdapter(new AzureJacksonAdapter())
                .withResponseBuilderFactory(new AzureResponseBuilder.Factory())
                .withInterceptor(new ProviderRegistrationInterceptor(credentials))
                .build(), subscriptionId);

```

File: azure/pom.xml [hunk 6]
```diff
@@ -146,6 +146,11 @@
       <artifactId>azure-mgmt-containerregistry</artifactId>
       <version>1.0.0-beta6-SNAPSHOT</version>
     </dependency>
+    <dependency>
+      <groupId>com.microsoft.azure</groupId>
+      <artifactId>azure-mgmt-documentdb</artifactId>
+      <version>1.0.0-beta6-SNAPSHOT</version>
```
Comment (jianghaolu) [inline azure/pom.xml]: 1.0.1-SNAPSHOT and elsewhere

Comment (alvadb) [inline azure/pom.xml]: done

File: azure-mgmt-documentdb/src/main/java/com/microsoft/azure/management/documentdb/implementation/DocumentDBManager.java (hunk 5 above)
Comment (alvadb) [inline azure-mgmt-documentdb/src/main/java/com/microsoft/azure/management/documentdb/implementation/DocumentDBManager.java]: done, except for line with withInterceptor.

**Pull request merged**